### PR TITLE
WL-3899 When in dropbox mode , it displays username with displayId …

### PR DESCRIPTION
**Pull request by surya-janani**
…for all the folders .

### CODE REVIEW

File: content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java [hunk 1]
```diff
@@ -4310,7 +4310,22 @@ public int compare(Group g0, Group g1) {
 			ContentCollection collection = ContentHostingService.getCollection(collectionId);
 			
 			ListItem item = ListItem.getListItem(collection, null, registry, need_to_expand_all, expandedCollections, items_to_be_moved, items_to_be_copied, 0, userSelectedSort, false, null);
-			
+
+			if(dropboxMode && item.getMembers()!= null && !item.getMembers().isEmpty()) {
```
Comment (buckett) [inline content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java:4314]: Does this mean for every dropbox listing we lookup every file name to see if there is a user with that name?

File: content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java [hunk 2]
```diff
@@ -4310,7 +4310,22 @@ public int compare(Group g0, Group g1) {
...
+					for (ListItem member:item.getMembers()) {
+						String memberIdArray[] = member.getId().split("/");
+						if(memberIdArray.length>1 && member.isCollection()) {
+							try {
+								String uniqueId = UserDirectoryService.getUser(memberIdArray[3]).getDisplayId();
```
Comment (buckett) [inline content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java:4319]: Is there a possible `ArrayOutOfBoundsException` here when `memberIdArray` has a length of 2?

**Comment by buckett** [comment]
I think you should be changing how dropbox generates the displayname property for all the folders it creates, rather than overriding the displayed name when listing files. This will create confusion for example, if a user selects Edit Details on the folder it will still show the original name in the display name field I think.

**Comment by surya-janani** [comment]
Completely Agree. Closing and deleting request. Making changes in kernel/BaseContentService

**Pull request closed without merging**